### ISSUE TITLE
mini_align step resulting in error (no -r)

### ISSUE THREAD

**Issue by spencerlong1** [nanoporetech/medaka#358]
Hi, I have been trying to use medaka on some assemblys but keep running into the same error, which looks like the following. I am new to the processing of nanopore sequences so sorry if this is a simple issue, but I am trying to follow various tutorials which never mention the -r at the minimap2 alignment, and cant find any other reference to it through googling. I simply input a flye output or a racon polished flye output, and am getting the following:

(medaka) 170-47:0203 sdl1c18$ medaka_consensus -i fastq_pass/nanofilt/0203_nanofilt_trimmed.fastq -d 0203_flye_output/assembly.fasta -o 0203_medaka_output/ -t 4 -m r941_min_high_g303
Checking program versions
This is medaka 1.0.3
Program    Version    Required   Pass     
bcftools   1.10.2     1.9        True     
bgzip      1.10.2     1.9        True     
minimap2   2.24       2.11       True     
samtools   1.10       1.9        True     
tabix      1.10.2     1.9        True     

Aligning basecalls to draft
mini_align [-h] -r <reference> -i <fastq>

Align fastq/a formatted reads to a genome using minimap2.

    -h  show this help text.
    -r  reference, should be a fasta file. If correspondng minimap indices
        do not exist they will be created. (required).
    -i  fastq/a input reads (required).
    -I  split index every ~NUM input bases (default: 16G, this is larger
        than the usual minimap2 default).
    -f  force recreation of index file.
    -a  aggressively extend gaps (sets -A1 -B2 -O2 -E1 for minimap2).
    -P  filter to only primary alignments (i.e. run samtools view -F 2308).
        Deprecated: this filter is now default and can be disabled with -A.
    -A  do not filter alignments to primary alignments, output all.
    -n  sort bam by read name.
    -c  chunk size. Input reads/contigs will be broken into chunks
        prior to alignment.
    -t  alignment threads (default: 1).
    -p  output file prefix (default: reads).
    -m  fill MD tag.
    -s  fill cs(=long) tag.
    -X  only create reference index files.
    -x  log all commands before running.
    -M  match score 
    -S  mismatch score
    -O  open gat penalty
    -E  extend gap penalty.
-r must be specified.
Failed to run alignment of reads to draft.

(medaka) 170-47:0203 sdl1c18$ 

Is there a fix for this? Am I missing something simple?

**Comment by cjw85** [comment]
Hi Spencer,

I'm not sure what's happening here, the medaka_consensus program should be passing your draft to the mini_align script (which is just a wrapper around minimap2) as the -r option. I do notice from the log that you have a fairly old version of medaka. Not that it should be broken, but could you first update to the latest version of medaka (I think we're on 1.5.*).

**Comment by spencerlong1** [comment]
thanks for the quick response. I have uninstalled and reinstalled via conda (simply using: "conda create -n medaka -c conda-forge -c bioconda medaka") and also tried "conda update medaka" but it is staying at 1.0.3. I'm using MacOS bash, and https://anaconda.org/bioconda/medaka seems to say conda installing for mac-osx.64 is still version 1.0.3 too. After doing this, as there is no change in the version, the above issue is still occurring. Also- Unsure if this is unusual or anything to do with the above issue but the package seems to take a lot longer than others I have used at the solving environment stage when downloaded via conda, although does eventually get there.

**Comment by cjw85** [comment]
Ah yes! bioconda stopped building the macOS package because it started breaking and neither we nor the bioconda developers could fix it. There are now only Linux packages available. (You might want to try installing with pip, which will attempt to compile the source code. You made need to install some extra things with homebrew for this to work however).

The older versions of medaka have a more complicated dependency chain than more recent versions, hence conda struggles to resolve the environment. I'd suggest switching to mamba.

As for your original issue I will have to did out the old code and check. Version 1.0.3 was current for a long time if I remember correctly so I'm surprised it's broken in such an obvious fashion.

**Comment by spencerlong1** [comment]
Okay, thank you. Will have a go with the recent linux versions and get back to you to see if it solves this or not, and thanks for the mamba recommendation, have been meaning to move on to it.

**Comment by spencerlong1** [comment]
Hi, finally got around to working on this again, sorry for the wait. I used my institutional linux cluster computer with latest version of medaka (still installed via conda) and and no such issue with the same data, so I think you are correct that is is an issue with the older macOS version before. I didn't get around to installing with pip on my mac simply because i got it to work with no issues straight away on the cluster, and I realised I could probably do with more power than my macbook anyway.

Thanks for the help
